### DISCUSSION TITLE
Isolate all audio-related code into its own C# package

### DISCUSSION THREAD

**Discussion by Speykious** [ppy/osu-framework#5864] [Ideas] [answered]
The idea is to put all audio-related classes into a separate csproj, perhaps called `osu.Framework.Audio`, and make sure that `osu.Framework` doesn't depend on `osu.Framework.Audio` to eliminate coupling. osu! and other apps/games would depend on both `osu.Framework` and `osu.Framework.Audio` if they want to have audio, while they would only depend on `osu.Framework` if they don't want to have audio.
This would make it easier to use osu!framework without a hard dependency on BASS, as its licensing is not the same as the rest.
Any kind of audio code would still be tightly coupled to BASS (until further refactoring in the future?), but the engine itself wouldn't be, and that means osu!framework could be used without any kind of audio.

I made a fork of osu!framework yesterday to try and remove all the audio code. It was very easy to do since any kind of audio code requires BASS, so when I remove the dependency on BASS, everything else has to follow. That's when I thought that maybe putting it all in its own package would be beneficial.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu-framework/issues/2784

tldr you'd need to do this yourself if you want it to happen. best you just fork and do it if you want, and if it ends up being useful to us we'll consider embracing it. chances are low though.

**Reply by Speykious** [reply]
I'll keep that in mind, thanks! Definitely attempting it.

I did look at this issue but my concern is separate, it's about being able to use osu!framework *without any audio* as opposed to just without BASS given the current coupling between the two. All audio code is coupled to BASS but it would be easy to decouple the audio code itself, is what I'm saying. So any refactoring that could let osu!framework support alternative audio engines would happen afterwards.

**Reply by peppy** [reply]
Then why split things out? Just made a startup flag to not start the audio thread or something.

Being blunt, we will not be splitting the audio namespace out into its own project as it's too much structural complexity.

**Reply by Speykious** [reply]
For me, the reason to split things out is BASS's licensing. I'd rather be sure that I'm not using BASS if I want to use osu!framework without audio, than having it sit in the code and potentially have to deal with its license.

**Reply by Speykious** [reply]
But maybe I'm missing something, for me it doesn't seem like that much added structural complexity - does it make it harder to iterate on new features or to do refactorings?

Well... If anything I'll try anyway and too bad it gets rejected. :/

**Reply by peppy** [reply]
just fork and remove if it bothers you IMO
